### PR TITLE
Extend rules

### PR DESCRIPTION
**Pull request by nickbattam**
@willrogers 
- extend BetweenRule
- add SelectionRule
- add name to base Rule

### CODE REVIEW

File: opimodel/rules.py [hunk 1]
```diff
@@ -21,3 +34,25 @@ def __init__(self, prop_id, pv, threshold):
...
         self._threshold = threshold
+
+
+class SelectionRule(Rule):
+
+    def __init__(self, prop_id, pv, options, var="pv0"):
+        """ Simple selection rule, e.g.:
+
+            widget.rules = []
+            options = [(-1, colors.INVALID), (1, colors.MAJOR), (2, colors.MINOR)]
```
Comment (willrogers) [inline opimodel/rules.py]: Should options be a dict?

File: opimodel/rules.py [hunk 2]
```diff
@@ -21,3 +34,25 @@ def __init__(self, prop_id, pv, threshold):
...
+            widget.rules.append(
+                rules.SelectionRule('on_color', pv_name, options, var="pvSev0"))
```
Comment (willrogers) [inline opimodel/rules.py]: Should have constants for pv0 and pvSev0.

File: opimodel/rules.py [hunk 3]
```diff
@@ -2,17 +2,30 @@
...
         self._prop_id = prop_id
+        if name is not None:
+            self._name = name
 
 
 class BetweenRule(Rule):
 
-    def __init__(self, prop_id, pv, min_val, max_val):
+    def __init__(self, prop_id, pv, min_val, max_val,
+             gt_equals=False, lt_equals=False):
```
Comment (willrogers) [inline opimodel/rules.py]: This is a good way to do it, but I can't quite find the right variable names here.

Comment (willrogers) [reply opimodel/rules.py]: It took me a few seconds to work out which way round this meant.

Comment (nickbattam) [reply opimodel/rules.py]: The names are ambiguous and were masking an implementation error. Will submit a patch to fix the names and rule writer

File: opimodel/rules.py (hunk 1 above)
Comment (nickbattam) [reply opimodel/rules.py]: Perhaps. The list ensures the order in which the cases are written, but that's not important for the equality comparisons we have implemented.

File: opimodel/rules.py (hunk 2 above)
Comment (nickbattam) [reply opimodel/rules.py]: Agreed

**Comment by nickbattam** [comment]
Incorrectly described branch content in PR: 'name' only added as attribute for SelectionRule. Default of 'Rule' used everywhere else.
Patching to add name for all Rules

**Pull request merged**